Horizon/Horizon/Sources/Features/Notebook/Common/View/View.swift

```diff
@@ -0,0 +1,145 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Core
+import SwiftUI
+import SwiftUICore
+
+struct NotesBody: View {
+    @State private var title: String
+
+    private let router: Router?
+
+    private let builder: () -> AnyView
+
+    private let backgroundColor = Color(hexString: "#FBF5ED")!
+
+    init(
+        title: String,
+        router: Router?,
+        @ViewBuilder _ builder: @escaping () -> some View
+    ) {
+        self.title = title
+        self.router = router
+        self.builder = { AnyView(builder()) }
+    }
+
+    var body: some View {
+        GeometryReader { geometry in
```

Comment:
We try to drift away from `GeometryReader` because it can cause unwanted behaviour in the layout as it fills all the available space. Since we are targeting iOS 17, it opens up a lot of new apis like `containerRelativeFrame`. I'd ask you to look into it to see if there's alternative solutions for setting the `contentMargins`.